Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,254 @@
+      this._debugLight.addChild(this._light);
+      this._debugLight.addChild(this._graphics);
+    }
+  }
+};
+
+gdjs.LightRuntimeObjectRenderer = gdjs.LightRuntimeObjectPixiRenderer; //Register the class to let the engine use it.
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.getRendererObject = function () {
+  // Mandatory, return the internal PIXI object used for your object:
+  if (this.DEBUG) {
+    return this._debugLight;
+  }
+  return this._light;
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function () {
+  if (this.DEBUG) this.updateGraphics();
+  this.updateVertexBuffer();
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.updateGraphics = function () {
+  var vertices = [this._object.x, this._object.y];
+  var raycast = this.raycastTest().reduce(function (acc, val) {
+    return acc.concat(val);
+  });
+  vertices.push.apply(vertices, raycast);
+
+  this._graphics.clear();
+  for (var i = 0; i < vertices.length; i += 2) {
+    var lineColor = i % 4 === 0 ? 0xff0000 : 0x00ff00;
+    this._graphics
+      .lineStyle(1, lineColor, 1)
+      .moveTo(vertices[0], vertices[1])
+      .lineTo(vertices[i], vertices[i + 1]);
+    if (i !== vertices.length - 2) {
+      this._graphics.lineTo(vertices[i + 2], vertices[i + 3]);
+    } else {
+      this._graphics.lineTo(vertices[2], vertices[3]);
+    }
+  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
+  this._light.shader.uniforms.center = new Float32Array([
+    this._object.x,
+    this._object.y,
+  ]);
+
+  var raycast = this.raycastTest();
+
+  var vertexBuffer = [this._object.x, this._object.y];
+
+  for (var i = 2; i < 2 * raycast.length + 2; i += 2) {
+    vertexBuffer[i] = raycast[i / 2 - 1][0];
+    vertexBuffer[i + 1] = raycast[i / 2 - 1][1];
+  }
+
+  var indexBuffer = [];
+
+  for (var i = 0; i < 3 * raycast.length; i += 3) {
+    indexBuffer[i] = 0;
+    indexBuffer[i + 1] = i / 3 + 1;
+    if (i / 3 + 1 !== raycast.length) indexBuffer[i + 2] = i / 3 + 2;
+    else indexBuffer[i + 2] = 1;
+  }
+
+  this._light.geometry
+    .getBuffer('aVertexPosition')
+    .update(new Float32Array(vertexBuffer));
+
+  this._light.geometry.getIndex().update(new Uint16Array(indexBuffer));
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
+  var result = [];
+  var centerX = this._object.x;
+  var centerY = this._object.y;
+  var halfOfDiag = Math.sqrt(2) * this._radius;
+
+  this._manager.getAllObstaclesAround(this._object, this._radius, result);
+  var hitBoxes = result.map(function (item) {
+    return item.owner.getHitBoxes();
+  });
+  hitBoxes.push(this._object.getHitBoxes());
+
+  var polygons = hitBoxes.reduce(function (acc, val) {
+    return acc.concat(val);
+  });
+  var vertices = polygons
+    .map(function (poly) {
+      return poly.vertices;
+    })
+    .reduce(function (acc, val) {
+      return acc.concat(val);
+    });
+
+  function _calculatePOI(angle) {
```

Comment:
To avoid having the JS engine re-create this function every time `raycastTest` is launched, let's instead make it "static" (I would name it something like: `gdjs.LightRuntimeObjectPixiRenderer.getClosestIntersectionVertex` or `gdjs.LightRuntimeObjectPixiRenderer.getClosestIntersectionPoint`)